geode-core/src/test/java/org/apache/geode/cache/query/internal/index/CompactRangeIndexTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.internal.index;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.geode.Statistics;
+import org.apache.geode.cache.RegionAttributes;
+import org.apache.geode.cache.query.internal.DefaultQueryService;
+import org.apache.geode.cache.query.internal.index.AbstractIndex.InternalIndexStatistics;
+import org.apache.geode.cache.query.types.ObjectType;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.internal.cache.CachePerfStats;
+import org.apache.geode.internal.cache.GemFireCacheImpl;
+import org.apache.geode.internal.cache.LocalRegion;
+import org.apache.geode.internal.cache.RegionEntry;
+
+public class CompactRangeIndexTest {
+
+  private CompactRangeIndex index;
+  LocalRegion region = mock(LocalRegion.class);
```

Comment:
Please change all fields to `private`. The access modifier should always be the most restrictive possible, and we should follow the same guidelines in tests as we do in product code.

Reply:
I also recommend you consider using the MockitoRule with STRICT_STUBS:
```
  @Rule
  public MockitoRule mockitoRule = MockitoJUnit.rule().strictness(STRICT_STUBS);
```
If you grep through Geode test code for this you'll find quite a few tests. The advantage of using this rule with STRICT_STUBS is that it will do some additional validating for you that really helps while creating or modifying the test. It will let you know if you have any mocks or when-clauses that are not necessary so you can delete them and streamline your test.

Reply:
Thanks for comments.

Reply:
Updated